.ci/test.sh

```diff
@@ -199,7 +199,12 @@ if [[ $TASK == "gpu" ]]; then
     if [[ $METHOD == "pip" ]]; then
         cd $BUILD_DIRECTORY && sh ./build-python.sh sdist || exit -1
         sh $BUILD_DIRECTORY/.ci/check_python_dists.sh $BUILD_DIRECTORY/dist || exit -1
-        pip install --user $BUILD_DIRECTORY/dist/lightgbm-$LGB_VER.tar.gz -v --install-option=--gpu || exit -1
+        pip install \
+            --user \
+            -v \
+            --install-option=--gpu \
+            $BUILD_DIRECTORY/dist/lightgbm-$LGB_VER.tar.gz \
+        || exit -1
```

Comment:
The changes here in `test.sh` aren't strictly related to the `CMakeLists.txt` changes.

This isn't actually event changing anything about this script. Just breaking these arguments onto individual lines, to make the size of the diff in #5759 a bit smaller (and therefore those changes a bit easier to review).